libeos-payg/manager.c

```diff
@@ -1060,6 +1209,8 @@ file_load_cb (GObject      *source_object,
 
               if (!epc_period_validate (used_code->period, &local_error))
                 {
+                  epg_multi_task_increment (task);
+                  g_task_set_task_data (task, g_steal_pointer (&invalid_data_error), (GDestroyNotify)g_error_free);
```

Comment:
This is pretty ugly; I suppose it's my fault for making `epg_multi_task_return_error()` return immediately rather than waiting for all the other operations to complete. But would it be so bad to call it here?

Reply:
I think it's better to keep the pattern consistent of returning from a task in the callback after the file is written/deleted. Otherwise I don't understand how you'd know in `file_load_delete_cb` whether you need to return or if an error has already been returned to the task.